runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1240,15 +1233,15 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 		uintptr_t headerSize = 0;
 		if (compressObjectReferences()) {
 			uintptr_t size = ((J9IndexableObjectContiguousCompressed *)arrayPtr)->size;
-			headerSize = static_cast<J9JavaVM*>(_omrVM->_language_vm)->sizeOfJ9IndexableObjectContiguousCompressed;
+			headerSize = _contiguousHeaderSize;
 			if (0 == size) {
-				headerSize = static_cast<J9JavaVM*>(_omrVM->_language_vm)->sizeOfJ9IndexableObjectDiscontiguousCompressed;
+				headerSize = _discontiguousHeaderSize;
 			}
 		} else {
 			uintptr_t size = ((J9IndexableObjectContiguousFull *)arrayPtr)->size;
-			headerSize = static_cast<J9JavaVM*>(_omrVM->_language_vm)->sizeOfJ9IndexableObjectContiguousFull;
+			headerSize = _contiguousHeaderSize;
 			if (0 == size) {
-				headerSize = static_cast<J9JavaVM*>(_omrVM->_language_vm)->sizeOfJ9IndexableObjectDiscontiguousFull;
+				headerSize = _discontiguousHeaderSize;
```

Comment:
there is an opportunity for minor code sharing:

```
		uintptr_t headerSize = _contiguousHeaderSize;
		uintptr_t size = 0;

		if (compressObjectReferences()) {
			size = ((J9IndexableObjectContiguousCompressed *)arrayPtr)->size;
		} else {
			size = ((J9IndexableObjectContiguousFull *)arrayPtr)->size;
		}

		if (0 == size) {
			headerSize = _discontiguousHeaderSize;
		}
```

Reply:
Ah yes for sure, and actually will just use some ternary operators to simplify it even further.